### PR TITLE
Carry patch to docker/docker in pkg/build/vendor that allows compilation against our version

### PR DESCRIPTION
**Pull request by smarterclayton**
This is a blocker for cutting the alpha tag.

@bparees

[merge]

### CODE REVIEW

File: pkg/build/Godeps/Godeps.json [hunk 1]
```diff
@@ -3,10 +3,18 @@
...
 	],
 	"Deps": [
 		{
+			"ImportPath": "github.com/Azure/go-ansiterm",
+			"Rev": "70b2c90b260171e829f1ebd7c17f600c11858dbe"
+		},
+		{
+			"ImportPath": "github.com/Azure/go-ansiterm/winterm",
+			"Rev": "70b2c90b260171e829f1ebd7c17f600c11858dbe"
+		},
```
Comment (bparees) [inline pkg/build/Godeps/Godeps.json]: why isn't it sufficient that origin itself already godeps this?  are the godeps not unioned together?

Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: Hrm - testing that

Comment (smarterclayton) [inline pkg/build/Godeps/Godeps.json]: You're right, rebasing

File: pkg/build/vendor/github.com/docker/docker/pkg/term/windows/ansi_reader.go [hunk 2]
```diff
@@ -142,14 +142,14 @@ func readInputEvents(fd uintptr, maxBytes int) ([]winterm.INPUT_RECORD, error) {
 
 // KeyEvent Translation Helpers
 
-var arrowKeyMapPrefix = map[uint16]string{
+var arrowKeyMapPrefix = map[winterm.WORD]string{
```
Comment (bparees) [inline pkg/build/vendor/github.com/docker/docker/pkg/term/windows/ansi_reader.go]: still picking up a vendored file change w/ no godeps change?

Comment (smarterclayton) [inline pkg/build/vendor/github.com/docker/docker/pkg/term/windows/ansi_reader.go]: This is an UPSTREAM commit that mirrors the patch we carry on the main vendor dir.

**Comment by smarterclayton** [comment]
Crash is unrelated [merge]

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
Force merging, this is ridiculous.

**Pull request merged**